publisher-sdk/src/main/java/com/criteo/publisher/privacy/ConsentData.kt

```diff
@@ -0,0 +1,46 @@
+/*
+ *    Copyright 2020 Criteo
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package com.criteo.publisher.privacy
+
+import com.criteo.publisher.annotation.OpenForTesting
+import java.util.concurrent.atomic.AtomicReference
+
+@OpenForTesting
+class ConsentData {
+
+  /**
+   * Consent can be in any of the following three states:
+   *
+   * <ul>
+   *  <li> CONSENT_GIVEN: it is known that consent has been given by the user</li>
+   *  <li> CONSENT_NOT_GIVEN: it is known that consent has not been given by the user</li>
+   *  <li> CONSENT_UNKNOWN: consent is not known yet (pending CDB request) </li>
+   * </ul>
+   */
+  enum class ConsentStatus {
+    CONSENT_GIVEN,
+    CONSENT_NOT_GIVEN,
+    CONSENT_UNKNOWN
+  }
+  private val valueRef = AtomicReference<ConsentStatus>(ConsentStatus.CONSENT_UNKNOWN)
+
+  fun getConsentStatus(): ConsentStatus = valueRef.get()
```

Comment:
Nit: this can be a kotlin `var` with `get()/set()`

Reply:
If I do that then I would need to expose the `AtomicReference`. I am finding it cleaner to expose the value directly. If I don't use an `AtomicReference`, then I would need to make it volatile. (the consent status is written/read on different threads within code that is not synchronized).

Reply:
The `valueRef` can stay `private` and a `var` could use it:
```
private val valueRef = atomic

var consentStatus: ConsentStatus
  get() = valueRef.get()
  set(value) = valueRef.set(value)
```